runtime/mainnet/src/lib.rs

```diff
@@ -1040,4 +1132,198 @@ mod tests {
+			let call = RuntimeCall::System(frame_system::Call::remark { remark: vec![] });
+			let fee = UNIT / 10;
+			let tip = UNIT / 2;
+			let fee_plus_tip = fee + tip;
+			let treasury_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get());
+			let maintenance_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get());
+			let who_balance = pallet_balances::Pallet::<Runtime>::free_balance(&who);
+			let dispatch_info = call.get_dispatch_info();
+
+			// NOTE: OnChargeTransaction functions expect tip to be included within fee
+			let liquidity_info =
+				<OnChargeTransaction as OnChargeTransactionT<Runtime>>::withdraw_fee(
+					&who,
+					&call,
+					&dispatch_info,
+					fee_plus_tip,
+					0,
+				)
+				.unwrap();
+			<OnChargeTransaction as OnChargeTransactionT<Runtime>>::correct_and_deposit_fee(
+				&who,
+				&dispatch_info,
+				&call.dispatch(RuntimeOrigin::signed(who.clone())).unwrap(),
+				fee_plus_tip,
+				0,
+				liquidity_info,
+			)
+			.unwrap();
+
+			let treasury_expected_balance = treasury_balance + (fee_plus_tip / 2);
+			let maintenance_expected_balance = maintenance_balance + (fee_plus_tip / 2);
+			let who_expected_balance = who_balance - fee_plus_tip;
+
+			assert!(treasury_balance != 0);
+			assert!(maintenance_expected_balance != 0);
+
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get()),
+				treasury_expected_balance
+			);
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get()),
+				maintenance_expected_balance
+			);
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&who),
+				who_expected_balance
+			);
+		})
+	}
+
+	#[test]
+	fn test_fees_and_tip_split() {
+		new_test_ext().execute_with(|| {
+			let fee_amount = 10;
+			let fee =
+				<pallet_balances::Pallet<Runtime> as frame_support::traits::fungible::Balanced<
+					AccountId,
+				>>::issue(fee_amount);
+			let tip_amount = 20;
+			let tip =
+				<pallet_balances::Pallet<Runtime> as frame_support::traits::fungible::Balanced<
+					AccountId,
+				>>::issue(tip_amount);
+			let treasury_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get());
+			let maintenance_balance =
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get());
+			DealWithFees::on_unbalanceds(vec![fee, tip].into_iter());
+
+			// Each to get 50%, total is 30 so 15 each.
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&TreasuryAccount::get()),
+				treasury_balance + ((fee_amount + tip_amount) / 2)
+			);
+			assert_eq!(
+				pallet_balances::Pallet::<Runtime>::free_balance(&MaintenanceAccount::get()),
+				maintenance_balance + ((fee_amount + tip_amount) / 2)
+			);
+		});
+	}
+
+	#[test]
+	fn treasury_balance_converter_is_set() {
+		assert_eq!(
+			TypeId::of::<<Runtime as pallet_treasury::Config>::BalanceConverter>(),
```

Comment:
Sticking all the treasury pallet tests in a treasury submodule means you can just import the Config type globally for the module and then avoid all the duplicated prefixes. Can also remove the repetition of the treasury_ prefixes to test names.

Just a suggestion. Counter argument is potential wrapping due to another level of indentation.

The only reason for these tedious suggestions is that the tests read cleanly, without all the noise of awkward syntax. This makes it quicker to review (imo) and hopefully allows for more focus on finding problems, which is the main point.

Reply:
https://github.com/r0gue-io/pop-node/pull/428/commits/6d078eb3e6c15af2facb1aca85d5dd8e62694be1